js/app/demo/app.jsx

```diff
@@ -110,12 +113,15 @@ define(['react', 'jsx!editor', 'jsx!messages', 'jsx!fixedCode', 'jsx!configurati
                 });
             }.bind(this))
         },
+        componentDidUpdate: function() {
+            window.localStorage.setItem('linterDemoState', JSON.stringify(this.state));
+        },
         render: function() {
             return (
                 <div className="container editorRow">
                     <div className="row">
                         <div className="col-md-7">
-                            <Editor onChange={this.handleChange} text="var foo = bar;" errors={this.state.messages} />
+                            <Editor onChange={this.handleChange} text={this.initialText} errors={this.state.messages} />
```

Comment:
Most likely if you add it to dynamic state, whenever you type a letter, React tries to reset it back to old state, then changes state to what you typed, then tries to reset it to new state, etc.